### DISCUSSION TITLE
SDK: When are invariants run?

### DISCUSSION THREAD

**Discussion by JasperSurmont** [cosmos/community#18] [Chain Builders (Cosmos SDK)] [answered]
[This](https://docs.cosmos.network/main/building-modules/invariants.html) explains what invariants in Cosmos are, what they are used for and how to implement them. However, I am not sure when these are actually checked? Is it after every block? After some time? Only during testing? etc.

**Comment by julienrbrt** [comment, marked as answer]
You have to run a node with invariant check enabled (f.e `simd start --inv-check-period 10`, which will check the invariants every 10 blocks)
